### DISCUSSION TITLE
Is there an example of how to use WinRT Interop RadialController from Win UI 3 to support Surface Dial ??

### DISCUSSION THREAD

**Discussion by victorchinn** [microsoft/WindowsAppSDK#2892] [Q&A] [answered]
Looking for a code example of how to call WinRT Interop IRadialController so I can use a Surface Dial with the Win UI 3 app in C#.   Windows Community Toolkitfor WinUI  has a Text Extension that is supposed to work with RadialController but that feature is currently broken with no projected bug fix date.  There is sample code for a Windows RT interop folderpicker Class: https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/winrt-com-interop-csharp but not for IRadialControllerInterop interface.

**Comment by DarranRowe** [comment, marked as answer]
Well, in the page that you referenced is a table with the interop interface with the C♯ interop class. If you notice, it is a namepsace followed by the class name. What this means is that in this case, you should be looking for Windows.UI.Input.RadialControllerInterop.
This means it should be as simple as:
```
var windowHandle = WinRT.Interop.WindowNative.GetWindowHandle(this);
var radialController = Windows.UI.Input.RadialControllerInterop.CreateForWindow(windowHandle);
```

**Reply by victorchinn** [reply]
ahh right !!  THANK YOU for that clarification !
